File: web-ui-with-angular-cli/src/app/patient-manager/cases/models/referral.ts
import { Record } from 'immutable';
import * as moment from 'moment';
import { Case } from '../models/case';
import { Room } from '../../../medical-provider/rooms/models/room';
import { Doctor } from '../../../medical-provider/users/models/doctor';
import { User } from '../../../commons/models/user';
import { LocationDetails } from '../../../medical-provider/users/models//location-details';
import { Company } from '../../../account/models/company';
import { ReferralDocument } from './referral-document';
import { Speciality } from '../../../account-setup/models/speciality';
import { Tests } from '../../../medical-provider/rooms/models/tests';

const ReferralRecord = Record({
    id: 0,
    caseId: 0,
    referringCompanyId: 0,
    referringLocationId: 0,
    referringUserId: 0,
    referredToCompanyId: 0,
    referredToLocationId: 0,
    referredToDoctorId: 0,
    referredToRoomId: 0,
    referredToSpecialtyId: 0,
    referredToRoomTestId: 0,
    note: '',
    referredByEmail: '',
    referredToEmail: '',
    referralAccepted: 0,
    room: null,
    case: null,
    referringUser: null,
    referringLocation: null,
    referringCompany: null,
    referredToDoctor: null,
    referredToLocation: null,
    referredToCompany: null,
    referralDocument: null,
    referredToSpecialty: null,
    referredToRoomTest: null,
    isDeleted: false,
    createByUserID: 0,
    createDate: null,
    updateByUserID: 0,
    updateDate: null
});

export class Referral extends ReferralRecord {

    id: number;
    caseId: number;
    referringCompanyId: number;
    referringLocationId: number;
    referringUserId: number;
    referredToCompanyId: number;
    referredToLocationId: number;
    referredToDoctorId: number;
    referredToRoomId: number;
    referredToSpecialtyId: number;
    referredToRoomTestId: number;
    note: string;
    referredByEmail: string;
    referredToEmail: string;
    referralAccepted: number;
    room: Room;
    case: Case;
    referringUser: User;
    referringLocation: LocationDetails;
    referringCompany: Company;
    referredToDoctor: Doctor;
    referredToLocation: LocationDetails;
    referredToCompany: Company;
    referralDocument: ReferralDocument;
    referredToSpecialty: Speciality;
    referredToRoomTest: Tests;
    isDeleted: boolean;
    createByUserID: number;
    createDate: moment.Moment;
    updateByUserID: number;
    updateDate: moment.Moment;

    constructor(props) {
        super(props);
    }
}

File: Midas/MIDAS.GBX/GBBusinessObjects/Referral/Referral.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MIDAS.GBX.BusinessObjects
{
    public class Referral : GbObject
    {
        [JsonProperty("caseId")]
        public int CaseId { get; set; }

        [JsonProperty("referringCompanyId")]
        public int ReferringCompanyId { get; set; }

        [JsonProperty("referringLocationId")]
        public int ReferringLocationId { get; set; }

        [JsonProperty("referringUserId")]
        public int ReferringUserId { get; set; }

        [JsonProperty("referredToCompanyId")]
        public int? ReferredToCompanyId { get; set; }

        [JsonProperty("referredToLocationId")]
        public int? ReferredToLocationId{ get; set; }

        [JsonProperty("referredToDoctorId")]
        public int? ReferredToDoctorId{ get; set; }

        [JsonProperty("referredToRoomId")]
        public int? ReferredToRoomId{ get; set; }

        [JsonProperty("note")]
        public string Note{ get; set; }

        [JsonProperty("referredByEmail")]
        public string ReferredByEmail{ get; set; }

        [JsonProperty("referredToEmail")]
        public string ReferredToEmail{ get; set; }

        [JsonProperty("referralAccepted")]
        public bool? ReferralAccepted{ get; set; }

        [JsonProperty("referredToDoctor")]
        public Doctor Doctor { get; set; }

        [JsonProperty("referringUser")]
        public User User { get; set; }

        [JsonProperty("referredToLocation")]
        public Location Location { get; set; }

        [JsonProperty("referringLocation")]
        public Location Location1 { get; set; }

        [JsonProperty("ReferredToCompany")]
        public Company Company { get; set; }

        [JsonProperty("referringCompany")]
        public Company Company1 { get; set; }

        [JsonProperty("referredToSpecialtyId")]
        public int? ReferredToSpecialtyId { get; set; }

        [JsonProperty("referredToRoomTestId")]
        public int? ReferredToRoomTestId { get; set; }

        [JsonProperty("room")]
        public Room Room { get; set; }

        [JsonProperty("case")]
        public Case Case { get; set; }

        [JsonProperty("roomTest")]
        public RoomTest RoomTest { get; set; }

        [JsonProperty("specialty")]
        public Specialty Specialty { get; set; }

        [JsonProperty("referralDocument")]
        public List<ReferralDocument> ReferralDocument { get; set; }
    }
}
